src/Microsoft.AspNet.Mvc.Razor/Precompilation/PrecompilationTagHelperTypeResolver.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.CodeAnalysis;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc.Razor.Precompilation
+{
+    /// <summary>
+    /// <see cref="TagHelperTypeResolver"/> used during Razor precompilation.
+    /// </summary>
+    public class PrecompilationTagHelperTypeResolver : TagHelperTypeResolver
+    {
+        private readonly object _assemblyLookupLock = new object();
+        private readonly Dictionary<string, IEnumerable<ITypeInfo>> _assemblyLookup
+            = new Dictionary<string, IEnumerable<ITypeInfo>>(StringComparer.OrdinalIgnoreCase);
```

Comment:
We sure all assembly lookups are case insensitive. I somewhat remember doing `@addTagHelper` required the exact casing of the assembly.